migrations/3_deploy_node_ingress_rules_contract.js

```diff
@@ -12,10 +12,22 @@ const rulesContractName = Web3Utils.utf8ToHex("rules");
 let nodeIngress = process.env.NODE_INGRESS_CONTRACT_ADDRESS;
 let retainCurrentRulesContract = AllowlistUtils.getRetainNodeRulesContract();
 
+async function logCurrentAllowlist(instance) {
+    let s = await instance.getSize();
+    console.log("\n<<< current NODE allowlist >>>");
+    for (i = 0; i < s; i ++) {
+        let x = await instance.getByIndex(i);
+        console.log("enode://" + x[0] + "@" + x[1] + ":" + x[2]);
+    }
+    console.log("<<< end of enode list >>>");
```

Comment:
maybe change to "end of current NODE allowlist"

Reply:
done